cmd/runner.go

```diff
@@ -0,0 +1,185 @@
+	ctx := context.Background()
+
+	err := r.flag.Validate()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	jwt, err := readJWTFromFile(jwtFilePath)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	vaultToken, err := vaultLogin(jwt, r.flag.VaultPolicy, r.flag.VaultAddr)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = ensureVaultTokenSecretExist(clientSet, vaultToken, r.flag.VaultTokenSecretName, r.flag.VaultTokenSecretNamespace)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func ensureVaultTokenSecretExist(clientSet *kubernetes.Clientset, vaultToken, vaultTokenSecretName, vaultTokenSecretNamespace string) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vaultTokenSecretName,
+			Namespace: vaultTokenSecretNamespace,
+			Labels: map[string]string{
+				label.ManagedBy: name,
+			},
+		},
+		Data: map[string][]byte{
+			"token": []byte(vaultToken),
+		},
+	}
+
+	_, err := clientSet.CoreV1().Secrets(vaultTokenSecretNamespace).Get(vaultTokenSecretName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		fmt.Printf("Creating secret %#q for vault token...\n", vaultTokenSecretName)
+		_, err = clientSet.CoreV1().Secrets(vaultTokenSecretNamespace).Create(secret)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		fmt.Printf("Secret %#q for vault token has been created.\n", vaultTokenSecretName)
+
+	} else {
+		fmt.Printf("Updating secret %#q for vault token...\n", vaultTokenSecretName)
+		_, err = clientSet.CoreV1().Secrets(vaultTokenSecretNamespace).Update(secret)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		fmt.Printf("Secret %#q for vault token has been updated.\n", vaultTokenSecretName)
+	}
+
+	return nil
+}
+
+func readJWTFromFile(filepath string) (string, error) {
+	fmt.Printf("Reading service account JWT from file %#q ...\n", filepath)
+
+	jwt, err := ioutil.ReadFile(filepath)
+
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	return string(jwt), nil
+}
+
+func vaultLogin(jwt, role, vaultAddr string) (string, error) {
+	fmt.Printf("Logging into vault Kubernetes backend...\n")
+
+	url := fmt.Sprintf("%s/%s", vaultAddr, vaultAuthEndpoint)
+	values := map[string]string{"jwt": jwt, "role": role}
+
+	jsonValues, err := json.Marshal(values)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonValues))
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // nolint
```

Comment:
No way around that?

Reply:
We would need to pass vault CA into template. But we don't have it in config. Also we've used it in the same way via vault client. It is just hidden there so we never cared about that